src/app/launcher/create-app/mission-runtime-createapp-step/mission-runtime-createapp-step.component.ts

```diff
@@ -143,17 +147,21 @@ export class MissionRuntimeCreateappStepComponent extends LauncherStep implement
       // FIXME: use a booster change event listener to do this
       // set maven artifact
       if (this.launcherComponent.flow === 'osio' && this.stepCompleted) {
-        const artifactTS: number = Date.now();
-        const artifactRuntime = this.launcherComponent.summary.runtime.id.replace(/[.\-_]/g, '');
-        const artifactMission = this.launcherComponent.summary.mission.id.replace(/[.\-_]/g, '');
-        this.launcherComponent.summary.dependencyCheck.mavenArtifact = `booster-${artifactMission}-${artifactRuntime}-${artifactTS}`;
+        this.launcherComponent.summary.dependencyCheck.mavenArtifact = this.createMavenArtifact();
       }
     }
     this.updateBoosterViewStatus();
   }
 
   // Private
 
+  private createMavenArtifact(): string {
+    const artifactTS: number = Date.now();
+    const artifactRuntime = this.launcherComponent.summary.runtime.id.replace(/[.\-_]/g, '');
+    const artifactMission = this.launcherComponent.summary.mission.id.replace(/[.\-_]/g, '');
+    return `booster-${artifactMission}-${artifactRuntime}-${artifactTS}`;
+  }
+
```

Comment:
maybe we can move this code to the Dependency checker service it already does something like this

Reply:
I prefer making this change in another PR, wdyt @edewit?